src/platform_impl/windows/drop_handler.rs

```diff
@@ -1,253 +1,174 @@
-use std::{
-    ffi::OsString,
-    os::windows::ffi::OsStringExt,
-    path::PathBuf,
-    ptr,
-    sync::atomic::{AtomicUsize, Ordering},
-};
-
-use winapi::{
-    ctypes::c_void,
-    shared::{
-        guiddef::REFIID,
-        minwindef::{DWORD, UINT, ULONG},
-        windef::{HWND, POINTL},
-        winerror::S_OK,
-    },
-    um::{
-        objidl::IDataObject,
-        oleidl::{IDropTarget, IDropTargetVtbl, DROPEFFECT_COPY, DROPEFFECT_NONE},
-        shellapi, unknwnbase,
-        winnt::HRESULT,
+use std::{ffi::OsString, os::windows::ffi::OsStringExt, path::PathBuf, ptr};
+
+use windows::{
+    runtime::implement,
+    Win32::{
+        Foundation::{DV_E_FORMATETC, HWND, POINTL, PWSTR},
+        System::{
+            Com::{IDataObject, DVASPECT_CONTENT, FORMATETC, TYMED_HGLOBAL},
+            Ole::{DROPEFFECT_COPY, DROPEFFECT_NONE},
+            SystemServices::CF_HDROP,
+        },
+        UI::Shell::{DragFinish, DragQueryFileW, HDROP},
     },
 };
 
+// FIXME: Remove hack
```

Comment:
The pre-built bindings currently don't work with the `implement` macro

Reply:
The problem is that both the rust and [windows metadata](https://github.com/microsoft/win32metadata) namespaces need to match, so that I can `implement` something. That is currently impossible with the pre-built bindings, since they [remove](https://github.com/microsoft/windows-rs/blob/c604d64d065eb615c7085eb64ca11fb96ee3cb9b/src/lib.rs#L16) the necessary `Windows` root namespace. Here I recreate that namespace.

Reply:
You can do something like this:

https://github.com/microsoft/windows-samples-rs/blob/4bd8b9dcbcad61a2defa9c5d2630ba406406e4e4/core_app/src/main.rs#L5

Reply:
Thanks for the excellent suggestion!